synapseclient/core/utils.py

```diff
@@ -242,6 +242,32 @@ def id_of(obj: typing.Union[str, collections.abc.Mapping, numbers.Number]) -> st
     raise ValueError("Invalid parameters: couldn't find id of " + str(obj))
 
 
+def validate_submission_id(submission_id: typing.Union[str, int]) -> str:
+    """
+    Ensures that a given submission ID is either an integer or a string that
+    can be converted to an integer. Version notation is not supported for submission
+    IDs, therefore decimals are not allowed.
+
+    Arguments:
+        submission_id: The submission ID to validate
+
+    Returns:
+        The submission ID as a string
+
+    Raises:
+        ValueError: if the submission ID is invalid
+
+    """
+    if isinstance(submission_id, int):
+        return str(submission_id)
+    elif isinstance(submission_id, str) and submission_id.isdigit():
+        return submission_id
+    else:
+        raise ValueError(
```

Comment:
Ah interesting, sorry if there was any miss-communication, what are your thoughts on doing something like...

```
str(int(submission_id))
```

Reply:
I see, thanks for clarifying! I don't like the idea of silently transforming user inputs, so this wasn't my initial thought, but I can definitely patch it this way and throw a warning message to the user saying "Your ID has been changed", just in case a user intentionally fed a float submission ID thinking submission IDs also support dot notation.

I'll make this update in the morning, thanks!

Reply:
Agreed, there are vulnerabilities when doing `str(int(234234))`.  There are a couple of places in the code base that do something like `str(int(...))`.  

The problem with using `int` is that a user can technically accidentally pass in: `str(int(234234.3435346))`, which returns '234234'.  At this moment, I think that is _ok_.  I like your idea of pushing a `logger.warning`.

Reply:
Changes made https://github.com/Sage-Bionetworks/synapsePythonClient/pull/1135/commits/aa632fbc3ffd8c7c9d42e64785f1134034effcd8